networking/podenv.go

```diff
@@ -154,6 +173,25 @@ func (e *podEnv) podNSPathSave() error {
 	return nil
 }
 
+func (e *podEnv) podNSDestroy() error {
+	if e.podNS == nil {
+		return nil
+	}
+	_ = e.podNS.Close() // Don't care if err here
+
+	if err := syscall.Unmount(e.podNS.Path(), unix.MNT_DETACH); err != nil {
+		// if already unmounted, umount(2) returns EINVAL
+		if !os.IsNotExist(err) && err != syscall.EINVAL {
```

Comment:
are you sure you don't need to assert this is an Errno first? (I'm not actually sure, might need testing) https://golang.org/pkg/syscall/#Errno